R/sumstats_qc.R

```diff
@@ -52,8 +52,10 @@ rss_basic_qc <- function(sumstats, LD_data, skip_region = NULL) {
     allele_flip$target_data_qced <- allele_flip$target_data_qced %>%
       filter(!(variant_id %in% skip_variant))
   }
-
-  sumstats_processed <- allele_flip$target_data_qced %>% arrange(pos)
+  # remove totally same variant names, which is common to see in lifted data.
```

Comment:
I dont understand this --- the output of allele_qc would have duplicates? How does it happen in the first place? Would it make more sense to process it right in the allele_qc function so anything else using allele_qc would not have this issue? 

Still I don't get why it happens in the first place. Sounds like something unusual and should be addressed at its root not by removing it after.

Reply:
This case is indeed unusual, the phenomenom is found in Thomas' case. 

What I found is after (the original version of) allele qc, there are some variants having exactly same variant name: chr, pos and alleles, but **different z scores**. Then it will lead to some bugs since we cannot decide on which z to use, and allele qc cannot effectively remove these variants since it's checking totally the same rows.  Thomas said it's due to the lift from hg19 to hg38, after mapping some different variants can end up in the same position, and if happen to have same alleles, which can be due to the higly duplicated region, will lead to results like this.

My suggestion to him is simply remove these variants that I think is reasonable to do so. But I agree, we can put this line in allele QC?

Reply:
Thanks @SamuelSun2000 this is clear to me know.  in that case in our alele QC we should detect the situation you decribe above, and throw a proper error message to prompt the users to make changes. Let's not just remove duplicates for them. They need to decide what to keep.